include/actionlib/client/client_goal_handle_imp.h

```diff
@@ -106,6 +106,10 @@ CommState ClientGoalHandle<ActionSpec>::getCommState() const
   }
 
   assert(gm_);
+  if (!gm_) {
+    ROS_ERROR_NAMED("actionlib", "Client should have valid GoalManager");
+    return CommState(CommState::DONE);
+  }
```

Comment:
If `gm_` is being checked that should happen at the beginning of the function. Otherwise line 94 would already :boom: